llarp/dht/messages/gotname.cpp

```diff
@@ -0,0 +1,55 @@
+#include <dht/messages/gotname.hpp>
+#include <lokimq/bt_serialize.h>
+
+namespace llarp::dht
+{
+  constexpr size_t NameSizeLimit = 128;
+
+  GotNameMessage::GotNameMessage(const Key_t& from, uint64_t txid, service::EncryptedName data)
+      : IMessage(from), result(std::move(data)), TxID(txid)
+  {
+    if (result.ciphertext.size() > NameSizeLimit)
+      throw std::invalid_argument("name data too big");
+  }
+
+  bool
+  GotNameMessage::BEncode(llarp_buffer_t* buf) const
+  {
+    const std::string nonce((const char*)result.nonce.data(), result.nonce.size());
```

Comment:
Encoding this as a string rather than an integer won't run into endian-ness issues?

Reply:
Nevermind, I'm dumb.